[ROCm] Misc updates to ROCm-specific ci_build files

parallelo opened the pull request:
Misc updates to the ROCm-specific `ci_build` files `Dockerfile.rocm` and `rocm/run_py3_core.sh`.  This is needed to support our Community Supported Build efforts.

poznano-amd committed:
Misc updates to the ROCm-specific ci_build files Dockerfile.rocm and rocm/run_py3_core.sh

whchung commented:
@tatianashp and @gunan, this is another PR to trim community build for ROCm path. Along with #25596 it'd be very helpful if you could expedite the review for this PR so we can make TensorFlow community supported builds for AMD ROCm platform online.

gunan commented on tensorflow/tools/ci_build/linux/rocm/run_py3_core.sh:
```diff
@@ -19,21 +19,81 @@ set -e
+    --build_tests_only --test_output=errors --local_test_jobs=${TF_GPU_COUNT} --config=opt \
+    --run_under=//tensorflow/tools/ci_build/gpu_build:parallel_gpu_execute -- \
+    //tensorflow/... -//tensorflow/compiler/... -//tensorflow/contrib/... \
+    -//tensorflow/python/distribute:distribute_coordinator_test \
```
Instead of a hardcoded list, can we add "no_rocm" or "no_amd" tag and just add that tag to the test_tag_filters flag here?

parallelo replied:
Thanks for the review.  Sounds like a cleaner strategy.  We'll give that a try.

gunan replied:
If this script already produces broken builds, Let's just change this file to add the tag filter here, then we can tag the targets gradually, piece by piece.
WDYT?

parallelo replied:
Good idea, that sounds reasonable.  

i.e. Use this PR, remove all of the extra excluded tests, but add the `-no_rocm` tag.  Then, create a follow-on PR that'll add those tags to each of the individual tests.

parallelo replied:
Does that look better now?

poznano-amd committed:
Update rocm/run_py3_core.sh

Update rocm/run_py3_core.sh to minimize the unit test exclusion list, to add a "-no_rocm" tag filter, and to rearrange the order of bazel flags.

parallelo commented:
@gunan - Made the commit.  Is that what you were thinking?  (the follow-on step will be to submit another PR with the `no_rocm` tags to the specific unit tests)

gunan commented:
Yes, looks great. Thanks!

whchung commented:
@hgadig FYI. none of the failures in those 3 failing targets seem to be related to this PR.

hgadig commented:
> @hgadig FYI. none of the failures in those 3 failing targets seem to be related to this PR.

Yes. We are looking into this and helping to get merged.
